resources/mediawiki.special/mediawiki.special.preferences.js

```diff
@@ -70,6 +85,10 @@ $legends.each( function( i, legend ) {
 	});
 	$li.append( $a );
 	$preftoc.append( $li );
+
+	/** Wikia Change start - Add a custom event per tab - @see UC-206 */
+	$(document).trigger($tabName + '-complete');
```

Comment:
I think this can be simplified a bit. Currently, this is firing an event per tab, which is unnecessary and could slow things down if there were a bunch of tabs. `$.each()` is synchronous, so you could just fire one event after it's done looping. 

What would simplify this more, though, would be to fire an event when the tab is actually clicked. That way, you could reduce the event binding in preferences.js to just one click event, instead of a load event and a click event.